server/src/mindustry/server/ServerControl.java

```diff
@@ -172,6 +172,22 @@ public ServerControl(String[] args){
             }
         });
 
+        Events.on(WaveEvent.class, e -> {
+            if(state.is(State.playing) && Config.enableAutosave.bool()){
+                Fi file = saveDirectory.child("autosave-" + Strings.capitalize(world.getMap().name()) + "-" + state.wave + "." + saveExtension);
+                
+                Core.app.post(() -> {
+                    try{
+                        SaveIO.save(file);
+                        info("&lbAutosave completed.");
+                    }catch(Exception ex){
+                        String msg = "Autosave failed with: " + ex.getMessage();
```

Comment:
`Log.err("Autosaving failed.", ex)` would be preferable. Exception messages can be `null`, and they usually don't tell you what went wrong.

Reply:
You are right